http-clients/src/test/java/com/palantir/remoting3/clients/UserAgentTest.java

```diff
@@ -26,19 +26,19 @@
     @Test
     public void testCorrectHeaderFormatWithInstanceId() {
         UserAgent baseUserAgent = UserAgent.of("service", "instanceId", "1.0.0");
-        assertThat(baseUserAgent.headerFormat()).isEqualTo("service/instanceId (1.0.0)");
+        assertThat(UserAgents.format(baseUserAgent)).isEqualTo("service/instanceId (1.0.0)");
 
-        UserAgent derivedAgent = UserAgent.append(baseUserAgent, "remoting", "2.0.0");
-        assertThat(derivedAgent.headerFormat()).isEqualTo("service/instanceId (1.0.0), remoting (2.0.0)");
+        UserAgent derivedAgent = baseUserAgent.merge(UserAgent.of("remoting", "2.0.0"));
+        assertThat(UserAgents.format(derivedAgent)).isEqualTo("service/instanceId (1.0.0), remoting (2.0.0)");
```

Comment:
why not have the formatting method in `UserAgent`?

Reply:
just want to keep the data objects small